### DISCUSSION TITLE
Is setting global variables in boot (with mixins) an anti-pattern

### DISCUSSION THREAD

**Discussion by robMolloy** [quasarframework/quasar#11141] [General - Components / Directives / etc]
I stumbled across this topic which asks how to provide a global variable to all components and no one gave an answer (even though it is marked as solved). 

[https://forum.quasar-framework.org/topic/1795/solved-impossible-set-global-variable-on-quasar-0-15/15](url)

I have an answer which works very well using a mixin but all the answers seem to suggest that; A) it's not possible without vuex & B) it may be an anti-pattern.

3 questions;
1) Is my way of doing things incorrect? 
2) Could someone explain why I shouldn't be doing it like this if it is incorrect?
3) Would it still be incorrect if I was using someVar on every page?

in src/boot/globalVars - make sure to add 'globalVars' to boot in quasar.conf.js
```
import { boot } from 'quasar/wrappers';

const someVar = true;
const mixin1 = { data: () => ({ someVar: someVar }) };

export default boot(({ app }) => {
  app.mixin(mixin1);
});

```

**Comment by hawkeye64** [comment]
It would be better to use Vue's `provide/inject` technology.

**Reply by robMolloy** [reply]
Thanks @hawkeye64, I haven't used that before, I'll have a look through the docs.[https://v3.vuejs.org/guide/component-provide-inject.html#working-with-reactivity](url)

It looks as if you can do something similar by just putting someVar on Vue.prototype

```
import Vue from 'vue';
Vue.prototype.someVar = someVar;
```

I'd be interested if someone could explain the advantages/differences between the three options, but clearly the most resource-efficient way would be to use provide/inject as you suggested.

**Reply by hawkeye64** [reply]
That's from Vue 2 docs. There is no more `prototype` in Vue 3. There is a different way to add to global though. 

https://v3.vuejs.org/guide/migration/global-api.html#vue-prototype-replaced-by-config-globalproperties

**Reply by robMolloy** [reply]
That's great, thanks a lot!
